manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -59,17 +62,34 @@ type FybrikApplicationReconciler struct {
 
 // Reconcile reconciles FybrikApplication CRD
 // It receives FybrikApplication CRD and selects the appropriate modules that will run
-// The outcome is a Plotter containing multiple Blueprints that run on different clusters
+// The outcome is a Plotter that describes the modules required, the clusters in which they should run, and the overall data flows.
 func (r *FybrikApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	log := r.Log.WithValues("fybrikapplication", req.NamespacedName)
+	log := r.Log.With().Str("fybrikapplication", req.NamespacedName.String()).Logger()
+
 	// obtain FybrikApplication resource
 	applicationContext := &api.FybrikApplication{}
 	if err := r.Get(ctx, req.NamespacedName, applicationContext); err != nil {
-		log.V(0).Info("The reconciled object was not found")
+		// TODO - Should this be fatal rather than error level?
+		log.Error().Err(err).Str(logging.FYBRIKAPPUUID, "Unknown").Msg("The reconciled object was not found")
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
+
+	// Add the globally unique identifier for the FybrikApplication instance to the annotations for use in logging
+	// Only need to do this once, and not on every update to this resource
+	uuid := utils.GetFybrikApplicationUUID(applicationContext)
+	if applicationContext.GetGeneration() == 1 {
+		annotations := applicationContext.GetAnnotations()
+		if annotations == nil {
+			annotations = map[string]string{}
+			annotations[utils.FYBRIKAPPUUID] = uuid
+			applicationContext.SetAnnotations(annotations)
+		} else {
+			annotations[utils.FYBRIKAPPUUID] = utils.GetFybrikApplicationUUID(applicationContext)
+		}
```

Comment:
- Missing a call to applicationContext.SetAnnotations?
- You already got the uuid in line 79, no need to recalculate

Reply:
Fixed